sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIAsyncClient.java

```diff
@@ -1596,4 +1354,77 @@ Mono<AudioTranslation> getAudioTranslationAsResponseObject(String deploymentOrMo
         AudioTranslationOptions audioTranslationOptions) {
         return getAudioTranslation(deploymentOrModelName, "filename", audioTranslationOptions);
     }
+
+    /**
+     * Generates text-to-speech audio from the input text.
+     * <p>
+     * <strong>Request Body Schema</strong>
+     * </p>
+     * <pre>{@code
+     * {
+     *     input: String (Required)
+     *     voice: String(alloy/echo/fable/onyx/nova/shimmer) (Required)
+     *     response_format: String(mp3/opus/aac/flac) (Optional)
+     *     speed: Double (Optional)
+     * }
+     * }</pre>
+     * <p>
+     * <strong>Response Body Schema</strong>
+     * </p>
+     * <pre>{@code
+     * BinaryData
+     * }</pre>
+     *
+     * @param deploymentOrModelName Specifies either the model deployment name (when using Azure OpenAI) or model name
+     * (when using non-Azure OpenAI) to use for this request.
+     * @param speechGenerationOptions A representation of the request options that control the behavior of a
+     * text-to-speech operation.
+     * @param requestOptions The options to configure the HTTP request before HTTP client sends it.
+     * @throws HttpResponseException thrown if the request is rejected by server.
+     * @throws ClientAuthenticationException thrown if the request is rejected by server on status code 401.
+     * @throws ResourceNotFoundException thrown if the request is rejected by server on status code 404.
+     * @throws ResourceModifiedException thrown if the request is rejected by server on status code 409.
+     * @return the response body along with {@link Response} on successful completion of {@link Mono}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<BinaryData>> generateSpeechFromTextWithResponse(String deploymentOrModelName,
+        BinaryData speechGenerationOptions, RequestOptions requestOptions) {
+        // modelId is part of the request body in nonAzure OpenAI
+        try {
+            BinaryData speechGenerationOptionsWithModelId
+                = addModelIdJson(speechGenerationOptions, deploymentOrModelName);
+            return this.openAIServiceClient != null
+                ? this.openAIServiceClient.generateSpeechFromTextWithResponseAsync(deploymentOrModelName,
+                    speechGenerationOptionsWithModelId, requestOptions)
+                : this.serviceClient.generateSpeechFromTextWithResponseAsync(deploymentOrModelName,
+                    speechGenerationOptionsWithModelId, requestOptions);
+        } catch (JsonProcessingException e) {
+            return Mono.error(e);
+        }
```

Comment:
We should use policy to make the switch between AOAI and OAI in next release. We should not to manually updating every generated method as this will not allow us to get updates from TypeSpec in the next generation.

Reply:
https://github.com/Azure/azure-sdk-for-java/issues/38970